core/drivers/crypto/caam/cipher/caam_cipher_mac.c

```diff
@@ -0,0 +1,883 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018-2020 NXP
+ *
+ * Implementation of Cipher MAC functions
+ */
+#include <caam_cipher.h>
+#include <caam_common.h>
+#include <caam_jr.h>
+#include <caam_utils_mem.h>
+#include <caam_utils_sgt.h>
+#include <caam_utils_status.h>
+#include <drvcrypt_mac.h>
+#include <mm/core_memprot.h>
+#include <string.h>
+#include <tee/cache.h>
+#include <utee_defines.h>
+
+#include "local.h"
+
+static TEE_Result do_update_mac(struct drvcrypt_cipher_update *dupdate);
+static TEE_Result do_update_cmac(struct drvcrypt_cipher_update *dupdate);
+
+/*
+ * Constant definitions of AES MAC algorithms
+ */
+static const struct cipheralg aes_mac_alg[] = {
```

Comment:
Why is this an array with two elements instead of two different globals? This seems a bit unsafe.

Reply:
I remove the arrays.